core/src/OpenACC/Kokkos_OpenACC_Team.hpp

```diff
@@ -0,0 +1,612 @@
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef KOKKOS_OPENACC_TEAM_HPP
+#define KOKKOS_OPENACC_TEAM_HPP
+
+#include <openacc.h>
+#include <impl/Kokkos_Traits.hpp>
+#include <impl/Kokkos_Spinwait.hpp>
+#include <Kokkos_Atomic.hpp>
+#include <OpenACC/Kokkos_OpenACC.hpp>
+#include <Kokkos_ExecPolicy.hpp>
+
+//----------------------------------------------------------------------------
+//----------------------------------------------------------------------------
+
+namespace Kokkos {
+namespace Impl {
+
+class OpenACCTeamMember {
+ public:
+  constexpr static int TEAM_REDUCE_SIZE      = 512;
+  constexpr static int DEFAULT_TEAM_SIZE_MAX = 512;
+  constexpr static int DEFAULT_TEAM_SIZE_REC = 128;
+
+  using execution_space      = Kokkos::Experimental::OpenACC;
+  using scratch_memory_space = execution_space::scratch_memory_space;
+
+  scratch_memory_space m_team_shared;
+  int m_team_scratch_size[2];
+  int m_team_rank;
+  int m_team_size;
+  int m_league_rank;
+  int m_league_size;
+  int m_vector_length;
+  int m_vector_lane;
+  int m_shmem_block_index;
+  void* m_glb_scratch;
+  void* m_reduce_scratch;
+
+ public:
+  KOKKOS_FUNCTION
+  const execution_space::scratch_memory_space& team_shmem() const {
+    return m_team_shared.set_team_thread_mode(0, 1, 0);
+  }
+
+  KOKKOS_FUNCTION
+  const execution_space::scratch_memory_space& team_scratch(int level) const {
+    return m_team_shared.set_team_thread_mode(level, 1,
+                                              m_team_scratch_size[level]);
+  }
+
+  KOKKOS_FUNCTION
+  const execution_space::scratch_memory_space& thread_scratch(int level) const {
+    return m_team_shared.set_team_thread_mode(level, team_size(), team_rank());
+  }
+
+  KOKKOS_FUNCTION int league_rank() const { return m_league_rank; }
+  KOKKOS_FUNCTION int league_size() const { return m_league_size; }
+  KOKKOS_FUNCTION int team_rank() const {
+#ifdef KOKKOS_COMPILER_NVHPC
+    return __pgi_vectoridx();
+#else
+    return m_team_rank;
+#endif
+  }
+  KOKKOS_FUNCTION int vector_length() const { return m_vector_length; }
+  KOKKOS_FUNCTION int team_size() const { return m_team_size; }
+  KOKKOS_FUNCTION void* impl_reduce_scratch() const { return m_reduce_scratch; }
+
+  // FIXME_OPENACC: OpenACC does not provide any explicit barrier constructs
+  // for device kernels.
+  KOKKOS_FUNCTION void team_barrier() const {
+    Kokkos::abort(
+        std::string(
+            "Kokkos::Experimental::OpenACC ERROR: OpenACC does not "
+            "provide any explicit barrier constructs for device kernels; exit!")
+            .c_str());
+    exit(0);
```

Comment:
Why do you have `exit`. You should never get here right?

Reply:
`exit(0)` is deleted.